src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1070,6 +1070,9 @@ public virtual void IntersectWith(IEnumerable<T> other)
             if (Count == 0)
                 return;
 
+            if (other == this)
+                return;
```

Comment:
Won't there still be a problem if you did, for example:
```C#
set.IntersectWith(set.Select(i => I));
```
?  In other words, still enumerating through 'this' but without reference equality between `other` and `this`?